[CI] ShardFollowTaskReplicationTests testRetryBulkShardOperations failing

**Build Scans:**
- [elasticsearch-periodic #4674 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/ruudntwh466d6)
- [elasticsearch-periodic-platform-support #4644 / amazonlinux-2_platform-support-aws](https://gradle-enterprise.elastic.co/s/gjis4k6u7pkae)
- [elasticsearch-periodic-platform-support #4644 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/jk6unag55r4iq)
- [elasticsearch-periodic-platform-support #4644 / rhel-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/lp4o4vplalb22)
- [elasticsearch-periodic-platform-support #4644 / windows-2016_checkpart2_platform-support-windows](https://gradle-enterprise.elastic.co/s/iu5qkh7lg5p4c)
- [elasticsearch-pull-request #39479 / part-2-windows](https://gradle-enterprise.elastic.co/s/pq6ng6grjmijg)
- [elasticsearch-pull-request #39451 / part-2-windows](https://gradle-enterprise.elastic.co/s/gw5e4qv5qorua)
- [elasticsearch-pull-request #39451 / part-2-windows](https://gradle-enterprise.elastic.co/s/a572l7tyseq7a)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ccr:test" --tests "org.elasticsearch.xpack.ccr.action.ShardFollowTaskReplicationTests.testRetryBulkShardOperations" -Dtests.seed=CF36D7484DC95307 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=ii-Yiii-CN -Dtests.timezone=Africa/Libreville -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ccr.action.ShardFollowTaskReplicationTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testRetryBulkShardOperations),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <131L>
     but: was <-1L>
```

**Issue Reasons:**
- [main] 8 failures in test testRetryBulkShardOperations (3.4% fail rate in 233 executions)
- [main] 3 failures in step part-2-windows (50.0% fail rate in 6 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.9% fail rate in 104 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -284,12 +284,6 @@ tests:
 - class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
   method: test {yaml=logsdb/10_settings/logsdb with default ignore dynamic beyond limit and subobjects false}
   issue: https://github.com/elastic/elasticsearch/issues/116054
-- class: org.elasticsearch.indices.state.CloseIndexIT
-  method: testConcurrentClose
-  issue: https://github.com/elastic/elasticsearch/issues/116073
-- class: org.elasticsearch.xpack.ccr.action.ShardFollowTaskReplicationTests
-  method: testRetryBulkShardOperations
-  issue: https://github.com/elastic/elasticsearch/issues/116080
 - class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
   method: test {yaml=logsdb/10_settings/logsdb with default ignore dynamic beyond limit and default sorting}
   issue: https://github.com/elastic/elasticsearch/issues/116062

